### PR TITLE
Support skip information reconfirm when using external IDP

### LINKED ISSUE #4177: Allow to configure ad/ldap identifier to map userid/domain to ks user account automatically
**What's it about?**

By configuring ldap/ad to KubeSphere, the console will show a confirm window when ldap/ad login for the first time, it's not user-friendly.
We'd better allow to configure ad/ldap identifier to map userid/domain to ks user account automatically.
**What's the reason why we need it?**

I believe this is an important feature for Kubesphere. There're a few use cases:

* every user in ldap/ad has to confirm that user name, it's not user-friendly.

Please leave your comments below if there's anyone agrees with me. Or just give me a thumb up.

**Area Suggestion**

/kind feature-request
/area iam

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <hongming@kubesphere.io>

### What type of PR is this?

/kind feature
/area auth

### What this PR does / why we need it:

Skip information confirmation will reduce user login steps when using a custom identity provider.

### Which issue(s) this PR fixes:

Fixes #4177

### Does this PR introduced a user-facing change?

```release-note
Support skip information confirmation when using external IDP
```

### CODE REVIEW

File: pkg/apiserver/authentication/oauth/oauth_options.go [hunk 1]
```diff
@@ -141,6 +142,11 @@ type IdentityProviderOptions struct {
 	//  - mixed:  A user entity can be mapped with multiple identifyProvider.
 	MappingMethod MappingMethod `json:"mappingMethod" yaml:"mappingMethod"`
 
+	// ConfirmationNotRequired means that when the user login successfully,
+	// reconfirm the account information is not required.
+	// Username from IDP must math [a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*
+	ConfirmationNotRequired bool `json:"confirmationNotRequired"`
```
Comment (RolandMa1986) [inline pkg/apiserver/authentication/oauth/oauth_options.go:148]: 1. A positive naming convention should be preferred.

Comment (wansir) [reply pkg/apiserver/authentication/oauth/oauth_options.go:148]: Compatible with the default configuration of the old version. Confirmation is required by default.

**Pull request merged**